presto-main/src/main/java/com/facebook/presto/sql/analyzer/Scope.java

```diff
@@ -58,13 +60,15 @@ private Scope(
             boolean queryBoundary,
             RelationId relationId,
             RelationType relation,
-            Map<String, WithQuery> namedQueries)
+            Map<String, WithQuery> namedQueries,
+            Map<Identifier, WindowSpecification> windowSpecifications)
     {
         this.parent = requireNonNull(parent, "parent is null");
         this.relationId = requireNonNull(relationId, "relationId is null");
         this.queryBoundary = queryBoundary;
         this.relation = requireNonNull(relation, "relation is null");
         this.namedQueries = ImmutableMap.copyOf(requireNonNull(namedQueries, "namedQueries is null"));
+        this.windowSpecifications = ImmutableMap.copyOf(requireNonNull(windowSpecifications, "windowSpecifications is null"));
```

Comment:
`requireNonNull` is not necessary here. `ImmutableMap.copyOf()` checks for nullness already